src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java

```diff
@@ -394,6 +394,92 @@ public Mono<FieldsDto> buildForm(
+     * @param resultUrl       the result url where the user should be redirected at
+     *                        the end of the payment session
+     * @param notificationUrl the notification url where notify the session
+     * @param cancelUrl       the url where the user should be redirected if the
+     *                        session is canceled by the user
+     * @param orderId         the orderId of the payment session
+     * @param customerId      the customerId url of the api
+     * @param paymentMethod   the payment method for which the form should be built
+     * @param defaultApiKey   default API key
+     * @param contractId      the wallet contractId
+     * @return An object containing sessionId and state
+     */
+    /*
+     * @formatter:off
+     *
+     * Warning java:S107 - Methods should not have too many parameters
+     * Suppressed because this method wraps the underlying API which has this many parameters
+     *
+     * @formatter:on
+     */
+    @SuppressWarnings("java:S107")
+    public Mono<FieldsDto> buildForm(
+                                     @NotNull UUID correlationId,
+                                     @NotNull URI merchantUrl,
+                                     @NotNull URI resultUrl,
+                                     @NotNull URI notificationUrl,
+                                     @NotNull URI cancelUrl,
+                                     @NotNull String orderId,
+                                     @NotNull String customerId,
+                                     @NonNull PaymentMethod paymentMethod,
+                                     @NonNull String defaultApiKey,
+                                     String contractId
+    ) {
+        return executeBuildForm(
+                correlationId,
+                merchantUrl,
+                resultUrl,
+                notificationUrl,
+                cancelUrl,
+                orderId,
+                customerId,
+                paymentMethod,
+                defaultApiKey,
+                contractId,
+                true
+        );
+    }
+
+    @SuppressWarnings("java:S107")
+    private Mono<FieldsDto> executeBuildForm(
+                                             @NotNull UUID correlationId,
+                                             @NotNull URI merchantUrl,
+                                             @NotNull URI resultUrl,
+                                             @NotNull URI notificationUrl,
+                                             @NotNull URI cancelUrl,
+                                             @NotNull String orderId,
+                                             @NotNull String customerId,
+                                             @NonNull PaymentMethod paymentMethod,
+                                             @NonNull String defaultApiKey,
+                                             String contractId,
+                                             boolean isSubsequentPayment
```

Comment:
is a subsequent payment when the contractId is present, this flag is not required

Reply:
Fixed with 9299a55e4ced37ef6988f4a87ba606769aa6d3be!